### PR TITLE
fix: render the participants in same order as they are created

### LINKED ISSUE #4946: The order of svg is reversed for sequence diagram participant nodes
### Description

Hi team we are using this package to support mermaid in Excalidraw. To compute the position and dimensions I am using the SVG rendered by mermaid. 
However when updating from `10.2.3` to `10.3.0` (so I can get the Actor Creation and Destruction supported 10.3.0 onwards), the order participant nodes is reversed and its only reversed for actor type Participant which makes the migration tricky.

Eg let's consider this diagram
The **indexes** in the diagram resembles the **order** of nodes on the **DOM**  

image

Further more the order when mixing actor symbols and participants is even more confusing

image

Was there any specific reason for this change ?
Also I would like to know if the team is open to storing the dimensions and coordinates when using the API `mermaid.mermaidAPI.getDiagramFromText` ? That would help us in not being dependent on SVG for calculating positions / dimensions.

### Steps to reproduce

Try in different versions (10.2.3 vs 10.3.0)

### Screenshots

Shared above

### Code Sample

_No response_

### Setup

- Mermaid version: 3.0.0
- Browser and Version: [Chrome, Edge, Firefox]

### Suggested Solutions

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by ad1992**
## :bookmark_tabs: Summary
Until v10.2.4, the order of DOM nodes was fine - which means that participant actors were rendered in same order as creation. However in https://github.com/mermaid-js/mermaid/commit/d06bb05c5fe7544ec2b34cfa71c788e15f27499d the order was reversed using `.lower()` which is confusing, hence I am reverting to the previous behaviour.

I am not sure if the change was intentional but I hope this won't break any behaviour and tests are passing

Additionally I have added one more example in docs with actor symbols for sequence diagram as there were none with actor symbols.

Resolves #4946

## :straight_ruler: Design Decisions

Keeping the order same as creation simplifies and makes it more intutive

### :clipboard: Tasks

Make sure you

- [x] :book: have read the [contribution guidelines](https://github.com/mermaid-js/mermaid/blob/develop/CONTRIBUTING.md)
- [ ] :computer: have added necessary unit/e2e tests.
- [ ] :notebook: have added documentation. Make sure [`MERMAID_RELEASE_VERSION`](https://github.com/mermaid-js/mermaid/blob/develop/packages/mermaid/src/docs/community/development.md#3-update-documentation) is used for all new features.
- [x] :bookmark: targeted `develop` branch

I tried to add a unit test but unfortunately this is dependent on `D3.js` hence its not working out, I will need to mock the implementation of `lower`correctly  to make it work.

### CODE REVIEW

File: demos/sequence.html [hunk 1]
```diff
@@ -174,7 +181,7 @@ <h1>Sequence diagram demos</h1>
         flowchart: { curve: 'basis' },
         gantt: { axisFormat: '%m/%d/%Y' },
         sequence: { actorMargin: 50 },
-        // sequenceDiagram: { actorMargin: 300 } // deprecated
+        // sequenceDiagram: { actorMargin: 300 }, // deprecated
```
Comment (ad1992) [inline demos/sequence.html:184]:
```suggestion
        // sequenceDiagram: { actorMargin: 300 } // deprecated
```

**Comment by ad1992** [comment]
@sidharthv96 let me know if you have any concerns with this change

**Comment by sidharthv96** [comment]
https://github.com/mermaid-js/mermaid/commit/d06bb05c5fe7544ec2b34cfa71c788e15f27499d#diff-1f8b0d46157ff209384aa6d146c2542142552e50ed6257ab525b1b789f5a0cde seems to have multiple places where `.lower()` is called. @Valentine14th, can you confirm if removing that won't be an issue?

**Comment by ad1992** [comment]
> [d06bb05#diff-1f8b0d46157ff209384aa6d146c2542142552e50ed6257ab525b1b789f5a0cde](https://github.com/mermaid-js/mermaid/commit/d06bb05c5fe7544ec2b34cfa71c788e15f27499d#diff-1f8b0d46157ff209384aa6d146c2542142552e50ed6257ab525b1b789f5a0cde) seems to have multiple places where `.lower()` is called. @Valentine14th, can you confirm if removing that won't be an issue?
> 
> @ad1992, a bit more descriptive commit messages than `fix` would help when debugging later on.

Are you referring to the individual commit messages in PR ? The main commit message communicates what is fixed in this PR (As shown in PR title). 
Since those individual commit messages will be squashed so I might have just committed it as `fix` for minor typos. I will reword it better going forward.

**Comment by sidharthv96** [comment]
We are not that strict with the commit messages. Squashing is usually left to the PR author's will without any interference while merging, unless there are lots of small, related commits in the PR. :)

**Comment by ad1992** [comment]
> We are not that strict with the commit messages. Squashing is usually left to the PR author's will without any interference while merging, unless there are lots of small, related commits in the PR. :)

I see got it :) 

Also regarding `.lower()` there is just one more place in `drawActorTypeActor` and removing that doesn't cause any issue (as per my testing) and all tests pass as well. Let me know If I should push that change or we can keep it for later as well in case we are unsure.

**Comment by sidharthv96** [comment]
Current
image
New
image

That's a problem @ad1992. 

image

**Comment by sidharthv96** [comment]
The unit tests won't find issues like these sadly. And our visual tests run only before release. So that's why all the tests passed.

**Comment by ad1992** [comment]
> Current image New image
> 
> That's a problem @ad1992.
> 
> image

Hmm interesting, this issue wasn't there in previous versions so probably something changed post that which is breaking it. will check

**Comment by ad1992** [comment]
I have pushed the fix and rendering the top actor with lines first which is intuitive as those are the first shapes to be drawn. 
The order is  = Top actor with lines -> messages -> bottom actors

However in the same PR https://github.com/mermaid-js/mermaid/commit/d06bb05c5fe7544ec2b34cfa71c788e15f27499d where ordering was changed, the top actors were pushed after messages as well which I have reverted here.

Let me know if this works well.

**Comment by ad1992** [comment]
Thank you! Any tentative date for the next release?

**Comment by aloisklink** [comment]
> Thank you! Any tentative date for the next release?

We're planning on making the next release a v11/major release, so it might take a bit longer than usual unfortunately. Assuming nothing changes, we're hoping it will be out before the end of the year :crossed_fingers:

**Comment by ad1992** [comment]
> > Thank you! Any tentative date for the next release?
> 
> We're planning on making the next release a v11/major release, so it might take a bit longer than usual unfortunately. Assuming nothing changes, we're hoping it will be out before the end of the year 🤞

Thanks for the update @aloisklink! I can see there is a [v10 milestone](https://github.com/mermaid-js/mermaid/milestone/11) so checking if there is any plan of v10  minor version before v11 ?

**Comment by aloisklink** [comment]
> Thanks for the update @aloisklink! I can see there is a [v10 milestone](https://github.com/mermaid-js/mermaid/milestone/11) so checking if there is any plan of v10 minor version before v11 ?

It looks like there might be a v10.6.2 patch release and this PR is in it: https://github.com/mermaid-js/mermaid/pull/5103 :crossed_fingers: You should be able to use a release candidate pre-release version already by adding [`"mermaid": "^10.6.2-rc.2"`](https://www.npmjs.com/package/mermaid/v/10.6.2-rc.2) to your `package.json` file.

**Comment by sidharthv96** [comment]
@ad1992 there was a regression. Do you think the fix will be easy?

reproduction: https://deploy-preview-5017--mermaid-js.netlify.app/edit#pako:eNp9kk1rwzAMhv-K6nM6WI4-dLSMsdsG7TEXzVYbU8fKHIculP732SZpS_bhi4X1vNIr4bNQrElI0dFnT07Rs8GDx6ZykM_aGkWwXK1gwx8SXslaTmEBNZ8APcHAPTxNeMxkOMskvBhHBYQa3TFxD7B2OgVXXnnCQNCiD0aZFl2Ad7b8s3t6je1NDhYzOarAHt4sPgJ26Z7yCc7ylEtyY6ZUpu-sxtKBr401dcHzkKmZmeXXWG0z0OKPNaUxHZ_-GzPWKGdub0VSclo2HtC4ua-M_mqszMZgHxfS3PxNuujubgFlFI2OdzUBOS0K0ZBv0Oj4Jc4JrUSoqaFKyBhq9MdKVO4SOewDbwenhAy-p0L0rY4zjt9HyD3aji7fu222Bw

**Comment by ad1992** [comment]
> @ad1992 there was a regression [#5188 (comment)](https://github.com/mermaid-js/mermaid/pull/5188#issuecomment-1889781800). Do you think the fix will be easy?
> 
> reproduction: https://deploy-preview-5017--mermaid-js.netlify.app/edit#pako:eNp9kk1rwzAMhv-K6nM6WI4-dLSMsdsG7TEXzVYbU8fKHIculP732SZpS_bhi4X1vNIr4bNQrElI0dFnT07Rs8GDx6ZykM_aGkWwXK1gwx8SXslaTmEBNZ8APcHAPTxNeMxkOMskvBhHBYQa3TFxD7B2OgVXXnnCQNCiD0aZFl2Ad7b8s3t6je1NDhYzOarAHt4sPgJ26Z7yCc7ylEtyY6ZUpu-sxtKBr401dcHzkKmZmeXXWG0z0OKPNaUxHZ_-GzPWKGdub0VSclo2HtC4ua-M_mqszMZgHxfS3PxNuujubgFlFI2OdzUBOS0K0ZBv0Oj4Jc4JrUSoqaFKyBhq9MdKVO4SOewDbwenhAy-p0L0rY4zjt9HyD3aji7fu222Bw

@sidharthv96 give me some time, I will look into it and confirm,

**Comment by ad1992** [comment]
So I looked into this and the changes which were made in the https://github.com/mermaid-js/mermaid/commit/d06bb05c5fe7544ec2b34cfa71c788e15f27499d does make sense, during actor creation and destroy its very tricky to determine the position of actors, I tried various ways but one or other is failing some case hence i think the author rendered the messages first and updated the position of actors in `adjustCreatedDestroyedData` and then actors were rendered.

my intention was to make the DOM rendering same as order of creation (which was earlier) but that looks hard and might increase complexities unnecessarily. Hence I would try to go the other way around and change the API instead so we can expose the x/y coords and dimensions as well via the `getDiagramFromText` API.

```
const diagram = await mermaid.mermaidAPI.getDiagramFromText(text);
diagram.db.getActors();
 ```
 
 Mainly the above `getActors` function should contain `rectData` (x, y, width, height, etc) so I can use that directly to compute the dimensions and position, currently `rectData` is null. @sidharthv96 let me know your thoughts :)
 
 I have a lot of more API ideas to make the API better which I would like to discuss as well.
 Also happy to move this to a separate issue if needed.

**Pull request merged**